doc/manual/src/design/overview.md

```diff
@@ -0,0 +1,13 @@
+# Overview
+
+Nix is broken into layers that operate fairly independently.
+
+At the top is the *command line interface*, i.e. the argument parsing of the various Nix executables.
+
+Below that is the Nix *expression language*, in which packages and configurations are written.
+These are the layers which users interact with most.
+
+Below that is the *store layer*, Nix's machinery for presenting and files and fully elaborated build plans, and also executing those build plans.
+The store layer may not be as visible, but this is the heart of Nix.
+
+This chapter will start there and work up towards the more user-facing interfaces described in the rest of the manual.
```

Comment:
My first reading of this sentence left me confused as to what "there" is referring to. I would find it easier to understand as follows:

"This chapter starts with the store layer and works towards the user-facing interfaces described in the rest of the manual."

I also think this is implying a statement of scope, but I don't quite understand what the scope is supposed to be. Perhaps the following is correct: 

"This chapter describes the store layer of nix and how it interacts with the expression language and the command line interfaces described in the other chapters in this manual."

Reply:
```suggestion
This chapter describes Nix starting with that bottom store layer, then working its way up until it reaches the more user-facing interfaces described in the rest of the manual."
```

Reply:
Does this take into account your points?